internal/controller/managedcluster_controller.go

```diff
@@ -297,13 +295,21 @@ func (r *ManagedClusterReconciler) updateCluster(ctx context.Context, mc *hmc.Ma
 		return ctrl.Result{}, nil
 	}
 
-	helmValues, err := setIdentityHelmValues(mc.Spec.Config, cred.Spec.IdentityRef)
-	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("error setting identity values: %w", err)
+	if err := mc.AddHelmValues(func(values map[string]any) error {
+		values["clusterIdentity"] = cred.Spec.IdentityRef
+
+		if _, ok := values["clusterLabels"]; !ok {
+			// Use the ManagedCluster's own labels if not defined.
```

Comment:
My understanding was that `clusterSelector` in `MultiClusterService` is supposed to refer hmc `ManagedCluster` instead of capi `Cluster` objects. In that case propagation should not be optional. Is my assumption valid?

Reply:
Right now the `ClusterSelector metav1.LabelSelector` field is fed directly into the the [`.spec.clusterSelector`](https://github.com/projectsveltos/addon-controller/blob/ebff58a741cbc9ca1e78383fd7e8c7d9cfda82c6/api/v1beta1/spec.go#L593) field for the Sveltos ClusterProfile object. So the label selectors are matching `Cluster` objects.